Anlab.Mvc/Services/IOrderService.cs

```diff
@@ -96,11 +99,11 @@ private TestDetails[] CalculateTestDetails(Order order)
             var allTests = order.GetTestDetails();
             // TODO: Do we really want to match on ID, or Code, or some combination?
             var selectedTestIds = orderDetails.SelectedTests.Select(t => t.Id);
-            var tests = PopulateSelectedTestsItemModel(selectedTestIds, allTests);
+            var tests = PopulateSelectedTestsItemModel(selectedTestIds, allTests).OrderBy(a => a.LabOrder);
 
             var calcualtedTests = new List<TestDetails>();
 
-            foreach (var test in orderDetails.SelectedTests)
+            foreach (var test in tests)

```

Comment:
@laholstege Is this correct?

Reply:
I'm thinking you want to keep the selectedTests, but add the LabOrder to the TestDetails model, then order by that in the return statement.

Reply:
@jSylvestre I want to keep LabOrder in the TestItemModel with RequestOrder. I ended up just removing dbTest because tests is already a list of TestItemModels that have the selected ids